core/src/test/resources/test/VerticalMultiline.stat

```diff
@@ -137,3 +140,130 @@ override def load(
+final case class SomeCaseClass[N](projectOrdering: N => R, someOtherString: String)(implicit ex: Context)
+>>>
+final case class SomeCaseClass[N](
+    projectOrdering: N => R,
+    someOtherString: String
+  )(implicit ex: Context)
+
+<<< class: if the type params fit in one line, make it so (multiple type param)
+final case class SomeCaseClass[N <: NodeUpperBound, TC[_], T <: SomeBoundUnderLimit: TC](projectOrdering: N => R, someOtherString: String)(implicit ex: Context)
+>>>
+final case class SomeCaseClass[
+    N <: NodeUpperBound,
+    TC[_],
+    T <: SomeBoundUnderLimit: TC
+  ](projectOrdering: N => R,
+    someOtherString: String
+  )(implicit ex: Context)
+
+<<< trait: if the type params fit in one line, make it so (single type param)
+private sealed trait SomeTrait[N]
+>>>
+private sealed trait SomeTrait[N]
+<<< trait: if the type params fit in one line, make it so (multiple type param)
+private sealed trait SomeTrait[N, TC[_], T <: SomeBoundUnderLimit]
+>>>
+private sealed trait SomeTrait[N, TC[_], T <: SomeBoundUnderLimit]
+
+<<< trait: multi-line for a single type param
+private sealed trait SomeTrait[ThisIsSomeVeryLongTypeParam <: SomeLongUpperBound] extends Something
+>>>
+private sealed trait SomeTrait[
+    ThisIsSomeVeryLongTypeParam <: SomeLongUpperBound]
+  extends Something
+
+<<< trait: multi-line for multiple params
+private sealed trait SomeTrait[ThisIsSomeVeryLongTypeParam <: SomeLongUpperBound, AndAnother] extends Something
+>>>
+private sealed trait SomeTrait[
+    ThisIsSomeVeryLongTypeParam <: SomeLongUpperBound,
+    AndAnother]
+  extends Something
+
+<<< Work with modifiers for the constructor
+final case class TypedId[A] private (value: Long)
+>>>
+final case class TypedId[A] private (value: Long)
+
+<<< Work with modifiers for the constructor (single line types)
+final case class SomeVeryLongName[SomeLongTypeName <: UpperBounds] private (value: Long, name: String)(implicit ex: Executor)
+>>>
+final case class SomeVeryLongName[SomeLongTypeName <: UpperBounds] private (
+    value: Long,
+    name: String
+  )(implicit ex: Executor)
+
+<<< Work with modifiers (private[this]) for the constructor (full multi-line)
+final case class SomeVeryLongName[SomeLongTypeName <: UpperBounds, SomeOtherTypeName] private[this](value: Long, name: String)(implicit ex: Executor)
+>>>
+final case class SomeVeryLongName[
+    SomeLongTypeName <: UpperBounds,
+    SomeOtherTypeName
+  ] private[this] (
+    value: Long,
+    name: String
+  )(implicit ex: Executor)
+
+<<< Work with modifiers (protected) for the constructor (full multi-line)
+final case class SomeVeryLongName[SomeLongTypeName <: UpperBounds, SomeOtherTypeName] protected(value: Long, name: String)(implicit ex: Executor)
+>>>
+final case class SomeVeryLongName[
+    SomeLongTypeName <: UpperBounds,
+    SomeOtherTypeName
+  ] protected (
+    value: Long,
+    name: String
+  )(implicit ex: Executor)
```

Comment:
It should not dangle because it creates an ugly result when combined with `extends` and `with` (unless we do some manipulation with those, but then it would need to ignore other settings and make the overall style more complex). We do dangle for defs though.